### PR TITLE
Connect with session ID

### PR DESCRIPTION
**Pull request by jburel**
Allow to connect using session id.
A way to test the PR
 * Run code in eclipse for example and print out the session ID
 * launch the application
 * Copy and paste this id in user field, password can be anything but not empty
 * Click connect.

Note: I might go for empty password @gusferguson what do you think?

### CODE REVIEW

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java [hunk 1]
```diff
@@ -1814,13 +1816,18 @@ client createSession(String userName, String password, String hostName,
...
+		
 		try {
 		    // client must be cleaned up by caller.
 			if (port > 0) secureClient = new client(hostName, port);
 			else secureClient = new client(hostName);
 			secureClient.setAgent(agentName);
-			ServiceFactoryPrx entryEncrypted
-			    = secureClient.createSession(userName, password);
+			ServiceFactoryPrx entryEncrypted;
+			if (userName.equals(password)) {
```
Comment (manics) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: What if a user's password is the same as their username

Comment (mtbc) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: omero/omero? :smiley:

Comment (sbesson) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: or any user created by our integration tests at the moment

Comment (joshmoore) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: `joinSession(userName)` == `createSession(userName, userName)` so this should be fine, but you could try using the guest connection to look for a user of this name if you want to be really careful.

**Comment by jburel** [comment]
I have to handle the case of username == password, (known) Just wanted to test it and mark it onhold

**Comment by jburel** [comment]
i also need to remove session ID from user field.

**Comment by jburel** [comment]
I will probably go ahead with @joshmoore suggestion and also empty password (or something like that)

**Comment by jburel** [comment]
guest option does not seem to work.

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java (hunk 1 above)
Comment (joshmoore) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: If guest isn't working, I really think this block or even the simpler `createSession(username, password)` is fine considering the implementation of `joinSession`:

https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/blitz/src/omero/client.java#L635

**Comment by gusferguson** [comment]
@jburel - I suspect the empty password would be the least confusing.

**Comment by manics** [comment]
Created a CLI session, used this to log in with Insight, works as expected.

**Comment by jburel** [comment]
@gusferguson: ok so I will review the log in screen.

**Comment by jburel** [comment]
Enable login control if no password. @manics: could you test it again? this time no password needed.

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java [hunk 2]
```diff
@@ -1814,13 +1813,29 @@ client createSession(String userName, String password, String hostName,
...
-			ServiceFactoryPrx entryEncrypted
-			    = secureClient.createSession(userName, password);
+		    if (port > 0) secureClient = new client(hostName, port);
+		    else secureClient = new client(hostName);
+		    secureClient.setAgent(agentName);
+		    ServiceFactoryPrx entryEncrypted;
+		    boolean session = true;
+		    try {
+		        //Check if it is a session first
+		        ServiceFactoryPrx guestSession = secureClient.createSession("guest", "guest");
```
Comment (aleksandra-tarkowska) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: isn't better to use ```getAdminService().getSecurityRoles()```

Comment (jburel) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java]: I am not connected at that point. That's the problem

**Comment by dominikl** [comment]
Doesn't work for me. With session id + empty password I get "Failed to login" message, session id + random string as password works.

**Comment by jburel** [comment]
@dominikl: thanks will check. I might have type something during my test

**Comment by jburel** [comment]
@dominikl: done

**Comment by manics** [comment]
It looks like the session-id is still saved in preferences- if you restart Insight the Username field is pre-populated with the last session-id.

**Comment by jburel** [comment]
@manics: ok thanks.

**Comment by jburel** [comment]
@manics: It is saved when the connection failed (it will always today)
If the connection is fine, then it should be kept. Only time we know it was a "session ID" credentials.

**Comment by dominikl** [comment]
Works like expected. Can confirm, if the connection could be established the session-id is not saved in the preferences, on a Insight restart the user name field is empty.
What surprised me, I can still connect to the session, after I closed the original Insight instance (started from eclipse), which created the session. Shouldn't it be closed at this point, does Insight not close the sessions properly?
Anyway, would be good if this could be merged soon, so I can rebase #3513 .

**Comment by jburel** [comment]
@dominikl: I discussed the session not being closed with @joshmoore.
This is outside the scope of this PR.
I have created a card. https://trello.com/c/NWT3qUma/430-client-del-session-still-alive

**Comment by jburel** [comment]
--no-rebase

**Pull request merged**